src/auth/auth.service.ts

```diff
@@ -19,21 +19,16 @@ export class AuthService {
   ) {}
 
   async register(createUserDto: CreateUserDto): Promise<SanitizedUser> {
-    if (!await this.usersService.validateEmail(createUserDto.email)) {
-      throw new HttpException("Registration error - this email is already used", HttpStatus.BAD_REQUEST);
-    }
-
-    if (!await this.usersService.validateUsername(createUserDto.username)) {
-      throw new HttpException("Registration error - this username is already used", HttpStatus.BAD_REQUEST);
+    try {
+      return await this.usersService.create(createUserDto);
     }
-
-    let user = await this.usersService.create(createUserDto);
-    
-    if (!await this.mailService.sendUserConfirmation(user.email, user.emailNonce)) {
-      throw new HttpException("Email could not be sent", HttpStatus.INTERNAL_SERVER_ERROR);
+    catch (err) {
+      if (err.code === 'P2002' && err.meta?.target?.length > 0) {
+        throw new HttpException(`Registration error - this ${err.meta.target[0]} is already used`, HttpStatus.BAD_REQUEST);
+      }
+      console.log(err);
+      throw new HttpException("Something went wrong", HttpStatus.INTERNAL_SERVER_ERROR, { cause: err });
```

Comment:
Please throw appropriate NestJS exception `InternalServerErrorException` from `@nestjs/common`.

Please **DO NOT** forward the raw errors to the consumers, as it might reveal sensitive information and we do not have control over those errors. The `cause` should always be hand-crafted or left as is for NestJS to interpret.

Reply:
Ok, got it.